experimental/personnames/src/pattern_regex_selector.rs

```diff
@@ -0,0 +1,465 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use lazy_static::lazy_static;
+use regex::{CaptureMatches, Captures, Match, Regex};
+
+use crate::api::{
+    FieldModifier, FieldModifierSet, NameField, NameFieldKind, PersonName,
+    PersonNamesFormatterError,
+};
+use crate::derive_core_prefix::handle_field_modifier_core_prefix;
+
```

Comment:
Hmm, so you are currently storing the name patterns as literal strings and then parsing them at runtime using the code in this file, right? Things like

```
"{given} {given2} {surname} {generation}{title}"
```

You should instead be using [icu_pattern](https://unicode-org.github.io/icu4x/docs/icu_pattern/index.html). However I think it's not completely ready yet; @zbraniecki do you have a suggestion on what Daniel should be doing here in the interim?